packages/compass-schema-validation/src/components/sample-documents/sample-documents.jsx

```diff
@@ -48,25 +61,19 @@ class SampleDocuments extends Component {
   static displayName = 'SampleDocuments';
 
   static propTypes = {
-    sampleDocuments: PropTypes.shape({
-      matching: PropTypes.object,
-      notmatching: PropTypes.object,
-      isLoading: PropTypes.bool,
-    }),
-  };
+    renderValidDocument: PropTypes.func,
+    renderInvalidDocument: PropTypes.func,
+  }
 
   /**
-   * Should the component update?
-   *
-   * @param {Object} nextProps - The next properties.
-   *
-   * @returns {Boolean} If the component should update.
+   * Instead of directly rendering these preview components
```

Comment:
I would like to bring the attention here to get some perspective on wether there is a better way or not. Stubbing an ES6 import had some [known issues](https://github.com/sinonjs/sinon/issues/1711#issuecomment-369238635).

Without that this component will try to render a connected component which will fail without a context. There was another idea of mocked store but that would probably be too much?

Reply:
This is not a pattern we use anywhere else and FWIW this creates a confusing component interface, I would suggest adding real interfaces that are used only for testing purposes as a last-ish resort. You can either provide a mocked store or create a real one in the testing environment, both options are valid and second one takes almost no effort and is a [recommended approach](https://redux.js.org/usage/writing-tests#guiding-principles) for the redux applications.

And on a separate note just generally looking at the test case you are introducing this for, I would suggest really looking at what we are testing there: all the tested elements are just rendered, so are we testing that React can render an HTML element in the DOM?

Reply:
Thanks for a quick one here. I wanted my tests to assert that there are two document previews and that was pretty much it. I thought of the mocked store approach but was not sure if that would be too much for such a little test. But I understand that this pattern diverts from standard practices, will switch to using a store then.

Reply:
Ahh I see your point about the test for SampleDocument component. There isn't any branching that could lead to a different render tree. Well that's right, given its current state its a dump component and the tests don't add much value. Thanks again @gribnoysup 👍

Reply:
Addressed the concern with the recent changes but have kept the test for the sake of it.

Reply:
Yeah, totally nothing wrong with keeping it, there is quite a lot of tests like that in monorepo right now unfortunately anyway. FWIW it is just a point to asses what's the amount of workarounds you'd want to add to actual source code to be able to test it, i.e. is it worth having those special test-only props to test that React will be able put stuff on the screen 😃